platform/qt/src/qmapboxgl.cpp

```diff
@@ -1007,26 +1007,27 @@ void QMapboxGL::removeAnnotation(QMapbox::AnnotationID id)
         \li QVariantList
     \endtable
 */
-void QMapboxGL::setLayoutProperty(const QString& layer, const QString& property_, const QVariant& value)
+bool QMapboxGL::setLayoutProperty(const QString& layer, const QString& property_, const QVariant& value)
 {
     using namespace mbgl::style;
 
     Layer* layer_ = d_ptr->mapObj->getStyle().getLayer(layer.toStdString());
     if (!layer_) {
         qWarning() << "Layer not found:" << layer;
-        return;
+        return false;
     }
 
-    if (conversion::setLayoutProperty(*layer_, property_.toStdString(), value)) {
-        qWarning() << "Error setting layout property:" << layer << "-" << property_;
-        return;
+    auto result = conversion::setLayoutProperty(*layer_, property_.toStdString(), value);
+    if (result) {
+        qWarning() << "Error setting layout property" << property_ << "on layer" << layer << ":" << QString::fromStdString(result->message);
     }
+    return result.operator bool();
```

Comment:
should we return inverted `result`? (since `result == true` means error)

Reply:
`return !result.has_value();`?

Reply:
While `has_value` is available in the [C++17 version of `std::optional`](https://en.cppreference.com/w/cpp/utility/optional), it is not available in the [experimental one](https://en.cppreference.com/w/cpp/experimental/optional), which is the one we use - see https://github.com/mapbox/mapbox-gl-native/blob/master/include/mbgl/util/optional.hpp.